Story#1359

adithyadinesh0412 opened the pull request.

adithyadinesh0412 committed:
init push

adithyadinesh0412 committed:
Merge branch 'develop' of https://github.com/ELEVATE-Project/survey-project-creation-service into story#1359

adithyadinesh0412 committed:
Merge branch 'develop' of https://github.com/ELEVATE-Project/survey-project-creation-service into story#1359

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

priyanka-TL commented on src/integration-tests/rollouts/responseSchema.js:
```diff
@@ -285,7 +285,139 @@ const getRolloutsListEmptyResponseSchema = {
+										},
+									],
+								},
+								gender: {
```
@adithyadinesh0412  can u please remove gender

priyanka-TL commented on src/integration-tests/rollouts/responseSchema.js:
```diff
@@ -285,7 +285,139 @@ const getRolloutsListEmptyResponseSchema = {
+									],
+								},
+							},
+							required: ['state', 'entity_targeting', 'district', 'block', 'gender', 'roles'],
```
@adithyadinesh0412  these all fields are not mandatory

priyanka-TL commented on src/integration-tests/rollouts/responseSchema.js:
```diff
@@ -285,7 +285,139 @@ const getRolloutsListEmptyResponseSchema = {
+			required: [
+				'title',
+				'resource_id',
+				'targeting_criteria',
```
@adithyadinesh0412  targeting_criteria, viewers is not mandatory

priyanka-TL commented on src/services/rollouts.js:
```diff
@@ -124,6 +125,98 @@ module.exports = class RolloutsHelper {
+					response.result &&
+					Object.keys(response.result).length > 0
+				) {
+					let resultData = response.result
```
@adithyadinesh0412  can u check with frontend team ? did they want the entity name in response?

adithyadinesh0412 replied:
Checked with FE team, They are Ok with only id in entity_targeting. @priyanka-TL

priyanka-TL reviewed (changes requested):
Reviewed on Dec 4th, 6PM

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

adithyadinesh0412 committed:
rollout list integration test added

priyanka-TL commented on src/services/rollouts.js:
```diff
@@ -124,6 +125,98 @@ module.exports = class RolloutsHelper {
+				type: common.ROLLOUT_TYPE_PROGRAM,
+			})
+
+			if (!rollout) {
```
@adithyadinesh0412  can u change this to !rollout?.id. because if any db error comes it will consider as rollout not empty

priyanka-TL commented on src/services/rollouts.js:
```diff
@@ -124,6 +125,98 @@ module.exports = class RolloutsHelper {
+					const userDetails = await this.fetchUserDetails([resultData.viewers])
+					const viewerUserIds = resultData.viewers
+
+					if (userDetails) {
```
@adithyadinesh0412  can u check the userDetails object is not empty

priyanka-TL commented on src/services/rollouts.js:
```diff
@@ -124,6 +125,98 @@ module.exports = class RolloutsHelper {
+					const viewerUserIds = resultData.viewers
+
+					if (userDetails) {
+						resultData.viewers = []
```
@adithyadinesh0412  why viewers array is defined inside ?

priyanka-TL commented on src/services/rollouts.js:
```diff
@@ -124,6 +125,98 @@ module.exports = class RolloutsHelper {
+					// fetch the org details from user service
+					const organizationDetails = await orgExtension.fetchOrganizationDetails([rollout.organization_id])
+
+					if (organizationDetails) {
```
@adithyadinesh0412  please check with status code and result

priyanka-TL reviewed (changes requested):
Reviewed on Dec 5th 7.26

priyanka-TL committed:
:q
Merge branch 'develop' of https://github.com/ELEVATE-Project/survey-project-creation-service into story#1359

priyanka-TL committed:
Merge branch 'develop' of https://github.com/ELEVATE-Project/survey-project-creation-service into story#1359

priyanka-TL committed:
rollout details

priyanka-TL committed:
postman updated

priyanka-TL committed:
optimized

aks30 commented on src/services/rollouts.js:
```diff
@@ -124,6 +125,83 @@ module.exports = class RolloutsHelper {
+				id: rolloutId,
+				organization_id: orgId,
+				user_id: loggedInUserId,
+				type: common.ROLLOUT_TYPE_PROGRAM,
```
@priyanka-TL why this filter?

priyanka-TL replied:
@aks30 removed, only for list its required

aks30 reviewed (changes requested):
Reviewed on 6 friday

priyanka-TL committed:
Pr comments addressed
